cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/triplestore/CgmesModelTripleStore.java

```diff
@@ -209,24 +209,40 @@ private void logNodeBreaker(boolean consideredNodeBreaker, Map<String, Boolean>
         }
     }
 
-    private Map<String, Boolean> computeEqModelHasEquipmentOperationProfile(PropertyBags modelProfiles) {
+    private Map<String, Boolean> computeModelHasOperationProfile(PropertyBags modelProfiles) {
+        // A bus/branch model with a single instance file where its node/breaker boundary has been assembled
+        // Must not be considered as node-breaker
         Map<String, Boolean> modelHasOperationProfile = new HashMap<>();
+        Map<String, Boolean> modelHasBoundaryOperationProfile = new HashMap<>();
         for (PropertyBag mp : modelProfiles) {
             String m = mp.get("FullModel");
             String p = mp.get(PROFILE);
             if (p != null) {
-                if (isEquipmentCore(p) || p.contains("/EquipmentBoundary/")) {
-                    modelHasOperationProfile.putIfAbsent(m, false);
-                }
-                if (isEquipmentOperation(p) || p.contains("/EquipmentBoundaryOperation/")) {
-                    modelHasOperationProfile.put(m, true);
-                    if (LOG.isInfoEnabled()) {
-                        LOG.info("Model {} is considered node-breaker", m);
-                    }
-                }
+                updateModelHasOperationProfile(modelHasOperationProfile, modelHasBoundaryOperationProfile, m, p);
             }
         }
-        return modelHasOperationProfile;
+        Map<String, Boolean> modelHasOperationProfile1 = new HashMap<>(modelHasOperationProfile);
+        modelHasBoundaryOperationProfile.forEach((m, v) -> modelHasOperationProfile1.merge(m, v, (vm, vbd) -> vm && vbd));
+        return modelHasOperationProfile1;
+    }
```

Comment:
I think we do not need  `modelHasOperationProfile1`, we can merge over `modelHasOperationProfile`

Reply:
you are right